rustls/src/msgs/deframer/buffers.rs

```diff
@@ -0,0 +1,115 @@
+use core::mem;
+use core::ops::Range;
+
+/// Conversion from a slice within a larger buffer into
+/// a `Range` offset within.
+#[derive(Debug)]
+pub(crate) struct Locator {
+    bounds: Range<*const u8>,
+}
+
+impl Locator {
+    pub(crate) fn new(slice: &[u8]) -> Self {
+        Self {
+            bounds: slice.as_ptr_range(),
+        }
+    }
+
+    pub(crate) fn locate(&self, slice: &[u8]) -> Range<usize> {
+        let bounds = slice.as_ptr_range();
+        debug_assert!(self.fully_contains(slice));
+        let start = bounds.start as usize - self.bounds.start as usize;
+        let len = bounds.end as usize - bounds.start as usize;
+        Range {
+            start,
+            end: start + len,
+        }
+    }
+
+    pub(crate) fn fully_contains(&self, slice: &[u8]) -> bool {
+        let bounds = slice.as_ptr_range();
+        bounds.start >= self.bounds.start && bounds.end <= self.bounds.end
+    }
+}
+
+/// Conversion from a `Range` offset to the original slice.
+pub(crate) struct Delocator<'b> {
+    slice: &'b [u8],
+}
+
+impl<'b> Delocator<'b> {
+    pub(crate) fn new(slice: &'b [u8]) -> Self {
+        Self { slice }
+    }
+
+    pub(crate) fn slice_from_range(&'_ self, range: &Range<usize>) -> &'b [u8] {
+        self.slice.get(range.clone()).unwrap()
```

Comment:
Is there a safety comment to add w.r.t to this `unwrap()`? It seems like it's safe with the assumption the `HandshakeDeframer` is the only source of `&Range<usize>` being provided to the `Delocator`, and only for spans fully contained within the larger slice?